wv2util/AppOverride.cs

```diff
@@ -191,601 +201,856 @@ public static void UpdateCollectionFromRegistry(ObservableCollection<AppOverride
-                        description += "registry (HKCU)";
-                        break;
-                    case StorageKind.HKLM:
-                        description += "registry (HKLM)";
-                        break;
-                    case StorageKind.EVCU:
-                        description += "environment variables (User)";
-                        break;
-                    case StorageKind.EVLM:
-                        description += "environment variables (Machine)";
-                        break;
-                }
-                return description;
-            }
-        }
-        
-        public RegistryKey RegistryRoot
-        {
-            get
-            {
-                switch (StorageKind)
-                {
-                    case StorageKind.HKCU:
-                        return Registry.CurrentUser;
-                    case StorageKind.HKLM:
-                        return Registry.LocalMachine;
-                    default:
-                        return null;
-                }
-            }
-        }
 
-        public void InitializationComplete()
-        {
-            m_InitializationComplete = true;
-        }
-        protected void OnPropertyChanged(string name)
-        {
-            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(name));
-            if (m_InitializationComplete)
-            {
-                if (name == "RegistryRoot")
-                {
-                    var removeRoot = this.StorageKind == StorageKind.HKCU ? Registry.LocalMachine : Registry.CurrentUser;
-                    AppOverrideList.RemoveEntryFromRegistry(this, removeRoot);
-                }
-                AppOverrideList.ApplyEntry(this);
-            }
-        }
-        public override string ToString()
-        {
-            return DisplayLabel;
-        }
+                }
+            }
+        }
+        private bool m_ReverseSearchOrder = true;
+
+        public bool IsEvergreenPreview
+        {
+            get => m_IsEvergreenPreview;
+            set
+            {
+                if (m_IsEvergreenPreview != value)
+                {
+                    m_IsEvergreenPreview = value;
+                    Properties.Settings.Default.IsEvergreenPreview = value;
+                    Properties.Settings.Default.Save();
+                    OnPropertyChanged("IsEvergreenPreview");
+                }
+            }
+        }
+        private bool m_IsEvergreenPreview = false;
 
-        public bool ReverseSearchOrder
+        public void LoadPersistedSettings()
         {
-            get => m_ReverseSearchOrder;
-            set
-            {
-                if (m_ReverseSearchOrder != value)
-                {
-                    m_ReverseSearchOrder = value;
-                    OnPropertyChanged("ReverseSearchOrder");
-                }
-            }
+          m_IsEvergreenPreview = Properties.Settings.Default.IsEvergreenPreview;

```

Comment:
Why is this a persisted setting for the app? Isn't this value per-override?

Reply:
Yes sorry, was experimenting with persisting. Updated to be in sync with existing reg key & env var values!